packages/collector/test/tracing/protocols/graphql/mockVersion.js

```diff
@@ -0,0 +1,45 @@
+/*
+ * (c) Copyright IBM Corp. 2022
+ * (c) Copyright Instana Inc. and contributors 2022
+ */
+
+'use strict';
+
+const semver = require('semver');
+const path = require('path');
+const mock = require('mock-require');
+const requireHook = require('../../../../../core/src/util/requireHook');
+
+const graphqlMajorDefault = semver.major(require(`${path.dirname(require.resolve('graphql'))}/package.json`).version);
+const GRAPHQL_VERSION = process.env.GRAPHQL_VERSION || graphqlMajorDefault.toString();
+const GRAPHQL_REQUIRE = GRAPHQL_VERSION === graphqlMajorDefault.toString() ? 'graphql' : `graphql-v${GRAPHQL_VERSION}`;
+process.env.GRAPHQL_REQUIRE = GRAPHQL_REQUIRE;
+
+if (GRAPHQL_REQUIRE !== 'graphql') {
+  mock('graphql', GRAPHQL_REQUIRE);
+
+  mock('graphql/language', `${GRAPHQL_REQUIRE}/language`);
+  mock('graphql/execution', `${GRAPHQL_REQUIRE}/execution`);
+  mock('graphql/utilities', `${GRAPHQL_REQUIRE}/utilities`);
+  mock('graphql/jsutils', `${GRAPHQL_REQUIRE}/jsutils`);
+  mock('graphql/validation', `${GRAPHQL_REQUIRE}/validation`);
+  mock('graphql/type', `${GRAPHQL_REQUIRE}/type`);
+  mock('graphql/polyfills', `${GRAPHQL_REQUIRE}/polyfills`);
+  mock('graphql/subscription', `${GRAPHQL_REQUIRE}/subscription`);
+  mock('graphql/error', `${GRAPHQL_REQUIRE}/error`);
+}
+
+if (GRAPHQL_REQUIRE === 'graphql-v16') {
+  mock('apollo-server-express', 'apollo-server-express-v3');
+}
+
+const originalOnFileLoad = requireHook.onFileLoad;
+requireHook.onFileLoad = function onFileLoad() {
+  if (arguments[0].toString() !== '/\\/graphql\\/execution\\/execute.js/') {
+    return originalOnFileLoad.apply(this, arguments);
+  }
+  const str = `/${GRAPHQL_REQUIRE}/execution/execute.js`;
+  const reg = new RegExp(str, '');
+  arguments[0] = reg;
+  return originalOnFileLoad.apply(this, arguments);
```

Comment:
Interesting solution. :) Maybe a comment would help, I needed a bit to wrap my head around this. Just to see if I am getting this correctly: We are effectively monkey-patching our own require-hook utility here to make it work despite the fact that we delegate all graphql-related require calls to different file system locations, right?

Reply:
Right:) You are right, hard to understand without comments. Will add some more comments